### PR TITLE
chore(EmotionFX): replace MCore::CreateFromAxisAndAngle with AZ::Quaternion::CreateFromAxisAngle

### PR DESCRIPTION
**Pull request by pollend**
Signed-off-by: Michael Pollind <mpollind@gmail.com>

## What does this PR do?

just replaces MCore::CreateFromAxisAndAngle with the equivalent method from AZ::Quaternion::CreateFromAxisAngle. both implements are basically the same the only main difference is MCore version normalizes the vector. not sure about the comment about this version being left handed and the AZCore version being right handed. I ended up writing a test to convince myself that both are equivalent after also comparing both implementations and seeing that they are the same with some subtle changes. There is also a small advantage with using simd operations with this change. 

```
class AngleTestFixture
        : public UnitTest::ScopedAllocatorSetupFixture {
        
    };

    TEST_F(AngleTestFixture, ToAngleTest)
    {
        std::uniform_real_distribution<float> unif;
        std::mt19937_64 rng(100);
        for(int x = 0; x < 1000; x++) {
            float angle = unif(rng);
            AZ::Vector3 v1 = AZ::Vector3(unif(rng), unif(rng), unif(rng)).GetNormalized();
            // MCore::CreateFromAxisAndAngle(v1, angle);
            AZ_Error("ToAngleTest", false, "angle1: %s angle2: %s", 
                AZStd::to_string(AZ::Quaternion::CreateFromAxisAngle(v1, angle)).c_str(),
                AZStd::to_string(MCore::CreateFromAxisAndAngle(v1, angle)).c_str()
                );
            
            EXPECT_THAT(MCore::CreateFromAxisAndAngle(v1, angle), IsClose(AZ::Quaternion::CreateFromAxisAngle(v1, angle)));
        }

    }
```

### CODE REVIEW

File: Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeAccumTransformNode.cpp
Comment (amzn-rhhong) [inline Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeAccumTransformNode.cpp:192]: Nit: You can do this with a conditional operator

**Review by amzn-rhhong — APPROVED** [review]

File: Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeAccumTransformNode.cpp
Comment (pollend) [reply Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeAccumTransformNode.cpp:192]: can you give me an example?

Comment (hultonha) [reply Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeAccumTransformNode.cpp:192]: I believe so, like this

```
const AZ::Quaternion targetRotation = axis.GetLengthSq() > 0.0f
    ? AZ::Quaternion::CreateFromAxisAngle(axis.GetNormalized(), MCore::Math::DegreesToRadians(360.0f * (inputAmount - 0.5f) * invertFactor))
    : AZ::Quaternion::CreateIdentity();
```

File: Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeTwoLinkIKNode.cpp
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Source/BlendTreeTwoLinkIKNode.cpp:483]: nit: Prefer breaking after assignment or paren

```
globalTransformA.m_rotation = AZ::Quaternion::CreateFromAxisAngle(
    axis.GetNormalized(), deltaAngle) *  globalTransformA.m_rotation;
```

**Review by hultonha — COMMENTED** [review]
The change seems okay to me but are there any tests to verify the functionality/change? Seems like this might cause some gnarly issues if we're not really careful. Be good to know more about how this was tested. Thanks!

**Comment by hultonha** [comment]
Also I don't see the test in the PR changed files 🤔

**Comment by pollend** [comment]
> The change seems okay to me but are there any tests to verify the functionality/change? Seems like this might cause some gnarly issues if we're not really careful. Be good to know more about how this was tested. Thanks!

Any clue how to test this in the editor. I'm not too strongly familiar with the EmotionFX editor. just tried to prove that both methods are the same.

**Comment by hultonha** [comment]
> > The change seems okay to me but are there any tests to verify the functionality/change? Seems like this might cause some gnarly issues if we're not really careful. Be good to know more about how this was tested. Thanks!
> 
> Any clue how to test this in the editor. I'm not too strongly familiar with the EmotionFX editor. just tried to prove that both methods are the same.

Probably a question for @amzn-jillich and @amzn-rhhong I'm afraid

**Comment by amzn-rhhong** [comment]
> > > The change seems okay to me but are there any tests to verify the functionality/change? Seems like this might cause some gnarly issues if we're not really careful. Be good to know more about how this was tested. Thanks!
> > 
> > 
> > Any clue how to test this in the editor. I'm not too strongly familiar with the EmotionFX editor. just tried to prove that both methods are the same.
> 
> Probably a question for @amzn-jillich and @amzn-rhhong I'm afraid

We have automation test for TwoLinkIKNode. If those test passes I think we can use it for validation. There hasn't been any test for the AccumTransformNode - if you'd like to follow up with some automated test for AccumTransformNode it would be a bonus point. I personally won't say that block this PR tho.

Strictly speaking for this PR we only need to validate if MCore::CreateFromAxisAndAngle and AZ::Quaternion::CreateFromAxisAngle yield the same correct result. Given that AZ::Quaternion math should be well tested and we are converting to AZ::Quaternion math, I'd say that is enough validation there.

**Comment by hultonha** [comment]
I've just kicked off a build now, might need another approval from @amzn-jillich and then hopefully will be good to go

**Review by amzn-jillich — COMMENTED** [review]
This is kind of the same situation as #10508. We really need to be careful with these types of math changes and need to verify by unit tests as well as test the features (in this case the AccumTransform and the TwoLinkIKNode) and post a video into the PR before we can approve this. If we don't spot issues early here, it will be really hard to find them later when we don't have a good idea anymore about what actually changed.

**Comment by pollend** [comment]
I set up an accumTransform and took a picture of the wireframe from the debug.

image

**Comment by amzn-jillich** [comment]
The AccumTransform node basically keeps accumulating and stores the rotation value for a given joint while the Transform node sets the input value directly. Here you can see an example of how that works:

https://user-images.githubusercontent.com/43751992/179551351-f6c07f65-789e-4231-a8c5-8aeb63423672.mp4

**Comment by pollend** [comment]
whats strange to me is the rotation axes from the preview is the Z axis when the rotation is suppose to be around the Y axis. unless the color for the bone axises are different?

Z: blue
X: red
Y: green

**Comment by amzn-jillich** [comment]
@pollend sounds good, so you are saying behavior is the same before and after the change but it looks like behavior was already around the wrong axis before? If yes, let's bring this change in and create another GHI for what you found.

Great work!

**Review by amzn-jillich — APPROVED** [review]

**Comment by pollend** [comment]
> @pollend sounds good, so you are saying behavior is the same before and after the change but it looks like behavior was already around the wrong axis before? If yes, let's bring this change in and create another GHI for what you found.
> 
> Great work!

The next question is is the viewport wrong or is the node in the tree wrong?

**Pull request merged**
